src/concrete/ml/torch/hybrid_model.py

```diff
@@ -53,35 +81,107 @@ def convert_conv1d_to_linear(layer_or_module):
-        self.private_q_module = None
-        self.fhe_local_mode = "disable"
-        self.client: Optional[FHEModelClient] = None
-        self.path_to_keys = None
-        self.path_to_client = None
-
-    def init_fhe_client(self, path_to_client: str, path_to_keys: str):
+        self.private_q_module: Optional[QuantizedModule] = None
+        # TODO: figure out if this is good
+        self.fhe_local_mode: FHEMode = FHEMode.CALIBRATE
+        self.clients: Dict[str, Tuple[str, FHEModelClient]] = {}
+        self.path_to_keys: Optional[Path] = None
+        self.path_to_clients: Optional[Path] = None
+        self.module_name: Optional[str] = module_name
+        self.model_name: Optional[str] = model_name
+        self.verbose = verbose
+
+    def init_fhe_client(
+        self, path_to_client: Optional[Path] = None, path_to_keys: Optional[Path] = None
+    ):  # pragma:no cover
         """Set the clients keys.
 
         Args:
             path_to_client (str): Path where the client.zip is located.
             path_to_keys (str): Path where keys are located.
+
+        Raises:
+            ValueError: if anything goes wrong with the server.
         """
-        # TODO: here we need to load fhe client.zip with FHEModelClient.
-        # Either by getting it from the server with the self.uid or
-        # directly getting it when downloading the model from HF.
+        self.path_to_clients = path_to_client
+        if self.path_to_clients is None:
+            self.path_to_clients = Path() / "clients"
+        self.path_to_clients.mkdir(exist_ok=True)
+        self.path_to_keys = path_to_keys
+        if self.path_to_keys is None:
+            self.path_to_keys = Path() / "keys"
+        self.path_to_keys.mkdir(exist_ok=True)
+
+        assert self.module_name is not None
+        shapes_response = requests.get(
+            f"{self.server_remote_address}/list_shapes",
+            data={"module_name": self.module_name, "model_name": self.model_name},
+        )
+        if shapes_response.status_code != 200:
+            # Add link to request content
+            raise ValueError(
+                f"Couldn't get shapes from server:\n{shapes_response.content.decode('utf-8')}"
+            )
+        shapes = shapes_response.json()
+        for shape in shapes:
+            client_response = requests.get(
+                f"{self.server_remote_address}/get_client",
+                data={
+                    "module_name": self.module_name,
+                    "model_name": self.model_name,
+                    "input_shape": shape,
+                },
+            )
+            if client_response.status_code != 200:
+                # Add link to request content
+                raise ValueError(
+                    f"Couldn't get client from server:\n{client_response.content.decode('utf-8')}"
+                )
+            path_to_client = self.path_to_clients / tuple_to_underscore_str(ast.literal_eval(shape))
```

Comment:
is this `ast.literal_eval(shape)` necessary ? isn't `shape` already some kind of string, which could be directly be given to `tuple_to_underscore_str` as an input ?

Reply:
https://docs.python.org/3/library/ast.html#ast.literal_eval

No attack possible other than memory exhaustion so pretty safe in our situation imho

Reply:
ok great, then fine for me !